src/java/org/apache/cassandra/utils/NoWaitRateLimiter.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.NotThreadSafe;
+import javax.annotation.concurrent.ThreadSafe;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Ticker;
+import com.google.common.math.LongMath;
+
+@ThreadSafe
+@SuppressWarnings("UnstableApiUsage")
+public class NoWaitRateLimiter
+{
+    @GuardedBy("this")
+    private Stopwatch stopwatch;
+
+    @GuardedBy("this")
+    private SmoothRateLimiter delegate;
+
+    private NoWaitRateLimiter(SmoothRateLimiter delegate, Ticker ticker)
+    {
+        this.delegate = delegate;
+        this.stopwatch = Stopwatch.createStarted(ticker);
+    }
+    
+    public static NoWaitRateLimiter create(double permitsPerSecond)
+    {
+        return create(permitsPerSecond, Ticker.systemTicker());
+    }
+
+    public static NoWaitRateLimiter create(double permitsPerSecond, Ticker ticker)
+    {
+        return create(permitsPerSecond, 1.0D, ticker);
```

Comment:
A 1 second burst made possible w/ stored permits is the default for the Guava `RateLimiter`, and I've simply carried that over here. We could make this configurable, although I don't have any compelling reason to do so yet.

Reply:
Might be worth mentioning here that if we keep this stored permits configuration (enough for a 1 second burst), it might be interesting to expose the number of currently available stored permits via JMX, etc. Thoughts?